XMonad/Layout/TallMastersCombo.hs

```diff
@@ -322,12 +322,10 @@ swapWindow w s =
       foc     = focus s
       downLst = down s
   in if w `elem` downLst
-     then let us   = takeWhile (/= w) downLst
-              d:ds = dropWhile (/= w) downLst
+     then let (us, nonEmpty -> d :| ds) = break (== w) downLst
               us'  = reverse us ++ d : upLst
           in  Stack foc us' ds
-     else let ds   = takeWhile (/= w) upLst
-              u:us = dropWhile (/= w) upLst
+     else let (ds, nonEmpty -> u :| us) = break (== w) upLst
```

Comment:
The first `nonEmpty` is safe but I don't think we know that `w` is in `upLst` if it's not in `downLst`

Reply:
Indeed!

427316b58dd423da049e675478ae8c0cf01b1816